frontend/src/bundles/common/components/transanction-table/components/future-transactions/future-transactions.tsx

```diff
@@ -0,0 +1,113 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import {
+    type Control,
+    type FieldErrors,
+    type FieldValues,
+} from 'react-hook-form';
+
+import { FaIcons } from '~/bundles/common/enums/fa-icons.enum';
+
+import { type ITransaction } from '../../types';
+import styles from '../styles.module.scss';
+import { Transaction } from '../transaction/transaction';
+
+interface FutureTransactionsProperties {
+    futureTransactions: ITransaction[];
+    futureTotals: number;
+    control: Control<FieldValues, null>;
+    errors: FieldErrors;
+}
+const FutureTransactions: React.FC<FutureTransactionsProperties> = ({
+    futureTransactions,
+    futureTotals,
+    control,
+    errors,
+}) => {
+    const [isShowFutureTransaction, setIsShowFutureTransaction] =
+        useState(false);
+
+    const handleToggleShowFutureTransactions = (): void => {
+        setIsShowFutureTransaction((previousState) => !previousState);
+    };
+    return (
+        <div className={styles.group}>
+            <div className={styles.transaction}>
+                <span className={styles.futureTitle}>
+                    Till the end of period
+                </span>
+            </div>
+            <div className={classNames(styles.totals, styles.transaction)}>
+                <div className={classNames(styles.columns)}>
+                    <FontAwesomeIcon
+                        className={styles.chevronIcon}
+                        icon={
+                            isShowFutureTransaction
+                                ? FaIcons.CHEVRON_DOWN
+                                : FaIcons.CHEVRON_UP
+                        }
+                        // eslint-disable-next-line react/jsx-no-bind
```

Comment:
?

Reply:
fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/192/commits/77b0aacb4b3e4c8404615f5670e0eaac4f165d78](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/url)